expression/builtin_time.go

```diff
@@ -3449,6 +3454,11 @@ func (b *builtinAddDateStringIntSig) evalTime(row chunk.Row) (types.Time, bool,
 		return types.ZeroTime, true, err
 	}
 
+	if invalidDate := date.InvalidZero(); invalidDate {
```

Comment:
I think it will help to remove the redundant check in the caller function, but it depends on the context to determine whether zeroDate is valid. For example, `adddate('0000-00-00', interval 1 day)`  is invalid becuase the first zero, while `adddate('2000-10-11', 0)` is valid with the second argument as a zero. It should be better leave to the caller function, but my current fix is not elegant.